src/transformer/events/core/course_viewed.php

```diff
@@ -21,8 +21,13 @@
 use src\transformer\utils as utils;
 
 function course_viewed(array $config, \stdClass $event) {
+    global $CFG;
+
     $repo = $config['repo'];
-    $user = $repo->read_record_by_id('user', $event->userid);
+
+    // Get relateduserid or a valid userid for guest.
+    $userid = $event->userid == 0 && isset($CFG->siteguest) ? $CFG->siteguest : $event->userid;
```

Comment:
@mlynn-lp 
Can the logic here be extracted out to a utility function?
It can then be reused elsewhere we encounter similar issues.